src/test/node_allocator_tests.cpp

```diff
@@ -0,0 +1,424 @@
+    }
+
+    // finally map_a is destroyed, so we got an entry back for mr_b.
+    CHECK_IN_RANGE(node_allocator::MemoryResourceTester::NumFreeAllocations(mr_a), 100U, 101U);
+    CHECK_IN_RANGE(node_allocator::MemoryResourceTester::NumFreeAllocations(mr_b), 1U, 2U);
+}
+
+// some structs with defined alignment and customizeable size
+
+namespace {
+
+template <size_t S>
+struct alignas(1) A1 {
+    char data[S];
+};
+
+template <size_t S>
+struct alignas(2) A2 {
+    char data[S];
+};
+
+template <size_t S>
+struct alignas(4) A4 {
+    char data[S];
+};
+
+template <size_t S>
+struct alignas(8) A8 {
+    char data[S];
+};
+
+template <size_t S>
+struct alignas(16) A16 {
+    char data[S];
+};
+
+template <size_t S>
+struct alignas(32) A32 {
+    char data[S];
+};
+
+} // namespace
+
+BOOST_AUTO_TEST_CASE(calc_required_allocation_size)
+{
+    static_assert(sizeof(A1<1>) == 1U);
+    static_assert(std::alignment_of_v<A1<1>> == 1U);
+
+    static_assert(sizeof(A16<1>) == 16U);
+    static_assert(std::alignment_of_v<A16<1>> == 16U);
+    static_assert(sizeof(A16<16>) == 16U);
+    static_assert(std::alignment_of_v<A16<16>> == 16U);
+    static_assert(sizeof(A16<24>) == 32U);
+    static_assert(std::alignment_of_v<A16<24>> == 16U);
+
+#if UINTPTR_MAX == 0xFFFFFFFFFFFFFFFF
+    static_assert(8U == node_allocator::detail::RequiredAllocationSizeBytes<A1<1>>());
+    static_assert(8U == node_allocator::detail::RequiredAllocationSizeBytes<A1<7>>());
+    static_assert(8U == node_allocator::detail::RequiredAllocationSizeBytes<A1<8>>());
+    static_assert(16U == node_allocator::detail::RequiredAllocationSizeBytes<A1<9>>());
+    static_assert(16U == node_allocator::detail::RequiredAllocationSizeBytes<A1<15>>());
+    static_assert(16U == node_allocator::detail::RequiredAllocationSizeBytes<A1<16>>());
+    static_assert(24U == node_allocator::detail::RequiredAllocationSizeBytes<A1<17>>());
+    static_assert(104U == node_allocator::detail::RequiredAllocationSizeBytes<A1<100>>());
+
+    static_assert(8U == node_allocator::detail::RequiredAllocationSizeBytes<A4<4>>());
+    static_assert(8U == node_allocator::detail::RequiredAllocationSizeBytes<A4<7>>());
+    static_assert(104U == node_allocator::detail::RequiredAllocationSizeBytes<A4<100>>());
+    static_assert(104U == node_allocator::detail::RequiredAllocationSizeBytes<A8<100>>());
+
+    static_assert(8U == node_allocator::detail::RequiredAllocationSizeBytes<A8<1>>());
+    static_assert(8U == node_allocator::detail::RequiredAllocationSizeBytes<A8<8>>());
+    static_assert(16U == node_allocator::detail::RequiredAllocationSizeBytes<A8<16>>());
+
+    static_assert(16U == node_allocator::detail::RequiredAllocationSizeBytes<A16<1>>());
+    static_assert(16U == node_allocator::detail::RequiredAllocationSizeBytes<A16<8>>());
+    static_assert(16U == node_allocator::detail::RequiredAllocationSizeBytes<A16<16>>());
+    static_assert(32U == node_allocator::detail::RequiredAllocationSizeBytes<A16<17>>());
```

Comment:
deduplicated in e2a5547f12e5da2718ba17ebac5d5aeb5a2e24d5. Also I have added a

```cpp
static_assert(__STDCPP_DEFAULT_NEW_ALIGNMENT__ >= std::alignment_of_v<T>, "make sure AllocatePool() aligns correctly");
```
Into `Allocate`, because actually the 16 byte alignments are not allowed when the pools are not correctly aligned as well.